drgn/cli.py

```diff
@@ -154,8 +164,6 @@ def main() -> None:
             )
         elif script_type == "elf":
             sys.exit(f"error: {args.script[0]} is a binary, not a drgn script")
-    else:
-        print(version, file=sys.stderr, flush=True)
```

Comment:
If we can, I'd like to keep printing the version as early as possible, so that if something goes horribly wrong (e.g., segfaults from C) in the initial setup of loading debug info, the output copied and pasted into bug reports will still include the version number.

Maybe we can print it here like before, and have the `banner_func` in `_main` remove the version header? Kinda ugly, but I think it'll work.

Reply:
What about making the `version_banner()` function a public API, and including in the documentation that we don't include it in the banner of `run_interactive()`?

Then, the main function is free to print it as early as possible so it's still there in case of a crazy error. Users that care to replicate the exact startup look of drgn can simply call `version_banner()` and print it early themselves, or add it in their `banner_func()`. 

When you think about it, while the other parts of the banner printed by `run_interactive()` seem objectively useful each time you load the interactive shell (e.g. to answer "which helpers are imported?"), the version banner seems to be something that would only be useful the first time. If I had a program with a command that allowed my users to drop into the drgn shell, I'd probably want to remove the version banner anyway.

Reply:
I like that idea.